src/external-server/external-server.ts

```diff
@@ -163,7 +164,7 @@ export abstract class ExternalServer {
 
         try {
             const response = await axios.post<T>(this.url, data, {
-                timeout: REQUEST_TIMEOUT,
+                timeout: this.hre.config.axios.timeout,
```

Comment:
Can't you dynamically import the hardhat runtime environment? Rather than having to modify all the constructors and propagate it.

Check other instances of doing `await import("hardhat")`

Reply:
indeed XD